### PR TITLE
Live change of theme from Preferences dropdown

### LINKED ISSUE #1048: Live change of theme after selecting from preferences dropdown
### Describe the request

The Arduino IDE should change theme immediately upon the user selecting a different theme from the "**Theme**" menu in the "**Preferences**" dialog.

If the user then clicks the <kbd>**Cancel**</kbd> button, the IDE should return to using the previously configured theme preference.

🙂 This will provide the user with an immediate preview of the appearance of the various available themes.

### Describe the current behavior

The Arduino IDE theme only changes after clicking the <kbd>**OK**</kbd> button in the "**Preferences**" dialog.

🙁 It is less convenient to evaluate the appearance of the various available themes.

### Arduino IDE version

2.0.0-rc7

### Operating system

Windows, Linux

### Operating system version

Windows 10, Ubuntu 20.04

### Issue checklist

- [X] I searched for previous requests in [the issue tracker](https://github.com/arduino/arduino-ide/issues?q=)
- [X] I verified the feature was still missing when using the latest [nightly build](https://github.com/arduino/arduino-ide#nightly-builds)
- [X] My request contains all necessary details

### PR DESCRIPTION
**Pull request by francescospissu**
### Motivation
The theme must be updated immediately if the user selects it from the Preferences drop-down menu. In this way it's possible to preview the appearance of the various themes available.

By pressing the Cancel button the initial theme must be restored.

### Change description
Restore the previous theme when the dialog is closed by pressing the Cancel button.

### Other information
This is an ad hoc solution for themes, in the long run this behavior should be generalized because other elements of the Preferences, such as the Editor font size or the Interface scale, may also require a live update to give immediate feedback to the user. In this case maybe a big rework of settings is needed.

Closes #1048.

### Reviewer checklist

* [ ] PR addresses a single concern.
* [ ] The PR has no duplicates (please search among the [Pull Requests](https://github.com/arduino/arduino-ide/pulls) before creating one)
* [ ] PR title and description are properly filled.
* [ ] Docs have been added / updated (for bug fixes / features)

### CODE REVIEW

**Review by per1234 — APPROVED** [review]
I gave it a try on Windows and it works perfectly for me. Very cool enhancement to the IDE.

Thanks Francesco!

**Review by kittaakos — APPROVED** [review]
It worked for me. I have verified it from the IDE2 started from the sources. 👍

File: arduino-ide-extension/src/browser/dialogs/settings/settings-component.tsx [hunk 1]
```diff
@@ -591,6 +591,9 @@ export class SettingsComponent extends React.Component<
     const theme = ThemeService.get().getThemes()[selectedIndex];
     if (theme) {
       this.setState({ themeId: theme.id });
+      if (ThemeService.get().getCurrentTheme().id !== theme.id) {
+        ThemeService.get().setCurrentTheme(theme.id);
```
Comment (davegarthsimpson) [inline arduino-ide-extension/src/browser/dialogs/settings/settings-component.tsx:595]: @kittaakos is the method `.setCurrentTheme` also called by Theia code after we click apply on the settings dialog?

**Comment by davegarthsimpson** [comment]
considering the proposed changes maybe we should mod [this](https://github.com/arduino/arduino-ide/blob/18264146186f47d5bbe97cd1281feb316f73eb99/arduino-ide-extension/src/browser/dialogs/settings/settings-component.tsx#L204) also:

```
  <select
    className="theia-select"
    value={
      ThemeService.get()
        .getThemes()
        .find(({ id }) => id === this.state.themeId)?.label ||
      nls.localize('arduino/common/unknown', 'Unknown')
    }
    onChange={this.themeDidChange}
  >
```

could be:

```
  <select
    className="theia-select"
   value={
      ThemeService.get()
        .getCurrentTheme()?.label ||
      nls.localize('arduino/common/unknown', 'Unknown')
    }
    onChange={this.themeDidChange}
  >
```

I don't see a reason to now reference the state value in the JSX (though maybe I'm missing something)

File: arduino-ide-extension/src/browser/dialogs/settings/settings-component.tsx (hunk 1 above)
Comment (kittaakos) [reply arduino-ide-extension/src/browser/dialogs/settings/settings-component.tsx:595]: Yes.

**Comment by francescospissu** [comment]
I agree @davegarthsimpson. I applied the change [here](https://github.com/arduino/arduino-ide/pull/1296/commits/9b0644ccf835d957e4352a6bc1d4b66357ec52ee), I tested it and it works fine.

**Review by davegarthsimpson — APPROVED** [review]

**Pull request merged**